### ISSUE TITLE
regex-tdfa-1.2.3.3 violates PVP contract

### ISSUE THREAD

**Issue by phadej** [ChrisKuklewicz/regex-tdfa#29]
```diff
-(=~~) x r = do let make :: (RegexMaker Regex CompOption ExecOption a, Monad m) => a -> m Regex
+(=~~) x r = do let make :: (RegexMaker Regex CompOption ExecOption a, Fail.MonadFail m) => a -> m Regex
```

**is** a more demanding type-signature (and thus breaking) change and might cause difficult to track compilation errors.

This feels minor, but the `1.2.3.3` should been `1.3`. You'll break pre-GHC-8.6 users which use `=~~` with not `MonadFail` monads.

**Comment by neongreen** [comment]
Hm, agreed. I released 1.3.0 and deprecated 1.2.3.3.

**Comment by phadej** [comment]
@neongreen thanks for prompt response!
